parser.go

```diff
@@ -99,6 +104,52 @@ func (amg *AuditMessageGroup) AddMessage(am *AuditMessage) {
 	}
 }
 
+// Find all `saddr=` occurrences in a message and do a lookup
+func (amg *AuditMessageGroup) mapDns(am *AuditMessage) {
+	data := am.Data
+	start := 0
+	end := 0
+
+	for {
+		if start = strings.Index(data, "saddr="); start < 0 {
+			break
+		}
+
+		// Progress the start point beyond the = sign
+		start += 6
+		if end = strings.IndexByte(data[start:], spaceChar); end < 0 {
+			end = len(data) - start
+			if end > 34 {
+				break
+			}
+		}
+
+		saddr := data[start : start+end]
+
+		var ip string
+
+		switch family := saddr[0:4]; family {
+		// 0200: ipv4
+		case "0200":
+			octet, _ := hex.DecodeString(saddr[8:16])
+			ip = fmt.Sprintf("%v.%v.%v.%v", octet[0], octet[1], octet[2], octet[3])
+		}
+
+		host, ok := c.Get(ip)
+		if ok {
+			amg.DnsMap[ip] = host.(string)
```

Comment:
There could be a performance enhancement here by separating ipv4 and ipv6 caches where ipv4 uses a proper uint32 and ipv6 uses a string. Not a today problem though.